verification/tagTCI.go

```diff
@@ -3,65 +3,227 @@
 package verification
 
 import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/sha512"
 	"errors"
+	"hash"
+
 	//"reflect"
 	"testing"
 )
 
 // This file is used to test the tagTCI command.
 
-func TestTagTCI(d TestDPEInstance, client DPEClient, t *testing.T) {
-	// Try to create the default context if isn't done automatically.
-	if !d.GetSupport().AutoInit {
-		handle, err := client.InitializeContext(InitIsDefault)
-		if err != nil {
-			t.Fatalf("Failed to initialize default context: %v", err)
-		}
-		defer client.DestroyContext(handle, DestroyDescendants)
+func TestTagTCI(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var err error
+	// Tag the default context
+	handle := getContextHandle(d, c, t, false)
+
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
 	}
 
+	digestLen := profile.GetDigestSize()
+
 	tag := TCITag(12345)
 	// Check to see our tag is not yet found.
-	if _, err := client.GetTaggedTCI(tag); !errors.Is(err, StatusBadTag) {
+	if _, err := c.GetTaggedTCI(tag); !errors.Is(err, StatusBadTag) {
 		t.Fatalf("GetTaggedTCI returned %v, want %v", err, StatusBadTag)
 	}
 
-	// Tag the default context
-	var ctx ContextHandle
+	// Make sure using wrong locality reports error
+	testTagTCIWrongLocality(d, c, t, handle)
+
+	// Make sure default handle returns same handle when extended
+	testTagTCIHandle(d, c, t, handle, digestLen, tag)
+
+	// Make sure the current TCI is updated
+	// For profiles which use auto-initialization, we don't know the expected TCIs so we can check only current TCI
+	testCurrentAndCumulativeTCI(d, c, t, handle, digestLen, tag)
 
-	handle, err := client.TagTCI(&ctx, tag)
+	// Make sure some other tag is still not found.
+	if _, err := c.GetTaggedTCI(TCITag(98765)); !errors.Is(err, StatusBadTag) {
+		t.Fatalf("GetTaggedTCI returned %v, want %v", err, StatusBadTag)
+	}
+
+	// Pass TCI input to DerivedChild context and check cumulative TCI
+	testForDerivedChildContexts(d, c, t, handle, digestLen, tag)
+}
+
+// Checks whether error is reported when the caller from one locality makes call to another locality.
+func testTagTCIWrongLocality(d TestDPEInstance, c DPEClient, t *testing.T, handle *ContextHandle) {
```

Comment:
Maybe we should just make each of these a separate test since they're testing difference cases.

Reply:
I shall add a BasicTestcase that checks common things like Wrong locality and Invalid handle errors

Reply:
Added the func and raised PR. link https://github.com/chipsalliance/caliptra-dpe/pull/255